includes/Storage/LogStorage.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace MediaWiki\Extension\Thanks\Storage;
+
+use Config;
+use DatabaseLogEntry;
+use ExtensionRegistry;
+use ManualLogEntry;
+use MediaWiki\Extension\Thanks\Hooks;
+use MediaWiki\Extension\Thanks\Storage\Exceptions\InvalidLogType;
+use MediaWiki\Extension\Thanks\Storage\Exceptions\LogDeleted;
+use MediaWiki\User\ActorNormalization;
+use RequestContext;
+use User;
+use Wikimedia\Rdbms\ILoadBalancer;
+
+class LogStorage {
+
+	protected ILoadBalancer $lb;
+	protected ActorNormalization $actorNormalization;
+
+	/**
+	 * @param ILoadBalancer $lb
+	 * @param ActorNormalization $actorNormalization
+	 */
+	public function __construct( ILoadBalancer $lb, ActorNormalization $actorNormalization ) {
+		$this->lb = $lb;
+		$this->actorNormalization = $actorNormalization;
+	}
+
+	public function thank( User $user, User $recipient, $uniqueId ) {
+		if ( !RequestContext::getMain()->getConfig()->get( 'ThanksLogging' ) ) {
+			return;
+		}
+		$logEntry = new ManualLogEntry( 'thanks', 'thank' );
+		$logEntry->setPerformer( $user );
+		$logEntry->setRelations( [ 'thankid' => $uniqueId ] );
+		$target = $recipient->getUserPage();
+		$logEntry->setTarget( $target );
+		$logId = $logEntry->insert();
+		$logEntry->publish( $logId, 'udp' );
+
+		if ( ExtensionRegistry::getInstance()->isLoaded( 'CheckUser' ) ) {
+			$recentChange = $logEntry->getRecentChange();
+			// TODO: This should be done in a separate hook handler
+			Hooks::updateCheckUserData( $recentChange );
+		}
+	}
+
+	/**
+	 * This checks the log_search data.
+	 */
+	public function haveThanked( User $thanker, string $uniqueId ): bool {
+		// TODO: Figure out why it's not getting the data from a replica
+		$dbw = $this->lb->getConnection( DB_PRIMARY );
+		$thankerActor = $this->actorNormalization->acquireActorId( $thanker, $dbw );
+		return $dbw->newSelectQueryBuilder()
+			->select( 'ls_value' )
+			->from( 'log_search' )
+			->join( 'logging', null, [ 'ls_log_id=log_id' ] )
+			->where(
+				[
+					'log_actor' => $thankerActor,
+					'ls_field' => 'thankid',
+					'ls_value' => $uniqueId,
+				]
+			)
+			->caller( __METHOD__ )
+			->fetchRow();
+	}
+
+	/**
+	 * @throws InvalidLogType
+	 * @throws LogDeleted
+	 */
+	public function getLogEntryFromId( Config $config, int $id ): ?DatabaseLogEntry {
```

Comment:
If we inject config vars via ServiceOptions then this also need not take a Config argument.